elasticdl_preprocessing/layers/to_sparse.py

```diff
@@ -0,0 +1,62 @@
+import tensorflow as tf
+
+
+class ToSparse(tf.keras.layers.Layer):
+    """Converts a `Tensor` to a `SparseTensor`, dropping ignore_value cells.
+    If the input is already a `SparseTensor`, just return it.
+
+    Example :
+    ```python
+        layer = ToSparse()
+        inp = tf.constant([["A", ""], ["B", "C"]], tf.string)
+        out = layer(inp)
+    ```
+    The expected output is
+    ```
+    tf.SparseTensor(
+            indices=np.array([[0, 0], [1, 0], [1, 1]]),
+            values=np.array(["A", "B", "C"]),
+            dense_shape=(2, 2),
+        )
+    ```
+
+    Arguments:
+        ignore_value: Entries in inputs equal to this value will be
+            absent from the output `SparseTensor`. If `None`, default value of
+            inputs dtype will be used ('' for `str`, -1 for `int`).
+
+    Input shape: A numeric or string `Tensor` of shape
+        `[batch_size, d1, ..., dm]`
+
+    Output shape: An `SparseTensor` with the same shape as inputs
+    """
+
+    def __init__(self, ignore_value=None):
+        super(ToSparse, self).__init__()
+        self.ignore_value = ignore_value
+
+    def call(self, inputs):
+        if isinstance(inputs, tf.SparseTensor):
+            return inputs
+        if self.ignore_value is None:
```

Comment:
This code snippet should be updated to:
```python
        if self.ignore_value is None:
            if inputs.dtype == tf.string:
                ignore_value = ""
            elif inputs.dtype.is_integer:
                ignore_value = -1
            else:
                ignore_value = -1
        else:
            ignore_value = self.ignore_value

        ignore_value = tf.cast(ignore_value, inputs.dtype)
        indices = tf.where(tf.not_equal(inputs, ignore_value))
```

Otherwise if use this layer in a keras model, exception will be thrown.

`self.ignore_value = tf.cast(self.ignore_value, inputs.dtype)`, cast receives a None input

Reply:
For other dtype, `ignore_value=0` may be more reasonable. If we convert a sparse tensor to dense, the values whose indices are not in `SparseTensor.indices` are zero.